cppcache/src/ThinClientRegion.cpp

```diff
@@ -1066,8 +1066,8 @@ GfErrType ThinClientRegion::removeNoThrow_remote(
 
   // do TCR remove
   TcrMessageDestroy request(new DataOutput(m_cacheImpl->createDataOutput()),
-                            this, keyPtr, cvalue, aCallbackArgument,
-                            m_tcrdm.get());
+                            this, keyPtr, cvalue, cvalue == nullptr,
```

Comment:
Is it necessary to put this parameter here since we are sending the value as the parameter just before it? Internally we can check for `nullptr` rather than putting it on the caller.

Reply:
Since the region->remove() and region->destroy() APIs all use TcrMessageDestroy, there's actually two types of Destroy messages that can be sent to the server. This parameter is needed to send the proper one based on whether the user is removing a key or a key and a value.

Reply:
Gotcha! Confusing...

`shared_ptr` converts to boolean, where `nullptr` converts to `false` and all other values to `true`. so `!cvalue` == `cvlaue == nullptr`

Reply:
Yup. C++11 allows using cvalue == nullptr, or !cvalue.